src/eppdperf/analysis.py

```diff
@@ -0,0 +1,171 @@
+    else:
+        print("Timeout reached. Not added to group: ", end="")
+        for ac in accounts:
+            if ac not in group_members:
+                print(ac.get_self_contact().addr, end=", ")
+        print()
+
+    begin = time.time()
+    counter = []
+    for ac in group_members:
+        counter.append(ac)
+    while time.time() < float(begin) + args.timeout and len(counter) > 0:
+        for ac in counter:
+            for chat in ac.get_chats():
+                if chat.get_name() == group.get_name():
+                    grp = chat
+            msgs = grp.get_messages()
+            for msg in msgs:
+                if msg.is_in_seen():
+                    continue
+                msg.mark_seen()
+                msgcontent = parse_msg(msg.text)
+                if msg.time_received is None or msgcontent.get("sender") == "spider":
+                    continue
+                origin = datetime.datetime(1970, 1, 1, tzinfo=datetime.timezone.utc)
+                msgreceived = (msg.time_received - origin).total_seconds()
+                duration = msgreceived - msgcontent.get("begin")
+                print("%s received message from %s after %.1f seconds" %
+                      (ac.get_self_contact().addr, msgcontent["sender"], duration))
+                output.submit_groupmsg_result(ac.get_self_contact().addr, msgcontent["sender"], duration)
+                if len(output.groupmsgs.get(ac.get_self_contact().addr)) == len(group_members) - 1:
+                    counter.remove(ac)
+
+    # send test messages to spider
+    testfilebytes = os.path.getsize(testfile)
+    if testfilebytes > 1024 * 1024:
+        testfilesize = str(round(testfilebytes / (1024 * 1024), 3)) + "MB"
+    else:
+        testfilesize = str(round(testfilebytes / 1024, 3)) + "KB"
+    print("Sending %s test file to spider from all accounts:" % (testfilesize,))
+    for ac in accounts:
+        if spider is not None:
+            if spider["addr"] is not ac.get_self_contact().addr:
+                check_account_with_spider(spac, ac, testfile)
+                ac.begin = time.time()
+
+    # wait until finished, or timeout
+    while len(accounts) > 0:
+        for ac in accounts:
+            if ac._shutdown_event.is_set():
+                accounts.remove(ac)
+            elif time.time() > ac.begin + args.timeout:
+                print("%d seconds timeout while waiting for echo to %s - test failed." %
+                      (args.timeout, ac.get_self_contact().addr))
+                output.submit_1on1_result(ac.get_self_contact().addr, "timeout", "timeout")
+                ac.shutdown()
+                ac.wait_shutdown()
+    if not args.yes:
+        answer = input("Do you want to delete all messages in the %s account? [y/N]" % (spider["addr"],))
+        if answer.lower() == "y":
+            for chat in spac.get_chats():
+                spac.delete_messages(chat.get_messages())
+        spac.shutdown()
```

Comment:
This if-condition is broken. If args.yes, the chats will *not* be deleted. This is a mess v.v

Reply:
Fixed in https://github.com/deltachat/eppdperf/pull/7/commits/e25769abe2ec60f3c23eb65faead1bd7767a9e03 :)